samcli/commands/build/build_context.py

```diff
@@ -68,6 +72,11 @@ def __enter__(self):
 
         self._build_dir = self._setup_build_dir(self._build_dir, self._clean)
 
+        if self._cached:
+            cache_path = pathlib.Path(self._cache_dir)
+            cache_path.mkdir(mode=0o755, parents=True, exist_ok=True)
```

Comment:
nit: Can we re-use property `_BUILD_DIR_PERMISSIONS` instead of hardcoding `0o755`?

Reply:
Yes, we can us it.